scribe-tool/scrawl.js

```diff
@@ -368,6 +371,47 @@
     scrawl.htmlFooter = footer;
   };
 
+  scrawl.preprocessLine = function(context, lines, lineI)
+  {
+    const line = lines[lineI];
+    const match = commentRx.exec(line);
+    if(!match)
+    {
+      return;
+    }
+    const [_, time, nick, msg] = match;
+
+    // check for a substitution
+    const replaceMatch = replaceRx.exec(msg);
+    if(replaceMatch)
+    {
+      const [_, cmd, delim, old, New, modifier] = replaceMatch;
+      if (cmd !== 's')
+      {
+        console.error(`command not supported on line ${lineI}: ${line}`);
+        return;
+      }
+      const maxReplaces = modifier === 'g' || modifier === 'G' ? Infinity : 1;
+      const endLineN = modifier === 'G' ? lines.length-1 : lineI-1;
+      let numReplaces = 0;
+      for(let i = endLineN; i >= 0; i--)
+      {
+        const line = lines[i];
+        const newLine = line.replace(old, New);
```

Comment:
```suggestion
        const newLine = line.replace(old, replacement);
```

Avoid using variable names that are also keywords in the language.